packagedb/to_purl.py

```diff
@@ -0,0 +1,75 @@
+#
+# Copyright (c) nexB Inc. and others. All rights reserved.
+# purldb is a trademark of nexB Inc.
+# SPDX-License-Identifier: Apache-2.0
+# See http://www.apache.org/licenses/LICENSE-2.0 for the license text.
+# See https://github.com/nexB/purldb for support or download.
+# See https://aboutcode.org for more information about nexB OSS projects.
+#
+
+from drf_spectacular.utils import OpenApiParameter
+from drf_spectacular.utils import extend_schema
+from packageurl.utils import get_golang_purl
+from rest_framework import routers
+from rest_framework import status
+from rest_framework import viewsets
+from rest_framework.response import Response
+
+from packagedb.serializers import GoLangPurlResponseSerializer
+from packagedb.serializers import GoLangPurlSerializer
+
+
+@extend_schema(
+    parameters=[
+        OpenApiParameter("go_package", str, "query", description="go import package"),
+    ],
+    responses={200: GoLangPurlResponseSerializer()},
+)
+class GolangPurlViewSet(viewsets.ViewSet):
+    """
+    Return a ``golang_purl`` from a standard go import string or
+    a go.mod string ``go_package``.
+    For example:
+    >>> get_golang_purl("github.com/gorilla/mux v1.8.1")
+    "pkg:golang/github.com/gorilla/mux@v1.8.1"
+    >>> # This is an example of go.mod string `package version`
+    >>> get_golang_purl("github.com/gorilla/mux")
+    "pkg:golang/github.com/gorilla/mux"
+    >>> #This is an example a go import string `package`
+    """
```

Comment:
We use Markdown to properly render the docstring for DRF Browsable API and Swagger API documentation.
```suggestion
    """
    Return a ``golang_purl`` from a standard go import string or
    a go.mod string ``go_package``.
    
    For example:

        >>> get_golang_purl("github.com/gorilla/mux v1.8.1")
        "pkg:golang/github.com/gorilla/mux@v1.8.1"
        >>> # This is an example of go.mod string `package version`
        >>> 
        >>> get_golang_purl("github.com/gorilla/mux")
        "pkg:golang/github.com/gorilla/mux"
        >>> #This is an example a go import string `package`
    """
```